cli/src/pcluster/config/mappings.py

```diff
@@ -1025,9 +1026,10 @@
             }),
             # Compute fleet
             ("compute_instance_type", {
+                "type": InstanceTypeCfnParam,
                 "default":
                     lambda section:
-                    "optimal" if section and section.get_param_value("scheduler") == "awsbatch" else "t2.micro",
+                    "optimal" if section and section.get_param_value("scheduler") == "awsbatch" else None,
```

Comment:
I think also the logic for the "optimal" value could be moved into the CfnParam. Of course we'd need a `ComputeInstanceTypeCfnParam` and a `MasterInstanceTypeCfnParam`.